KappaAnalysis/interface/Producers/ValidJetsProducer.h

```diff
@@ -187,6 +187,66 @@ class ValidJetsProducerBase: public KappaProducerBase, public ValidPhysicsObject
 		LambdaNtupleConsumer<KappaTypes>::AddFloatQuantity("sixthJetMass", [settings](KappaEvent const& event, KappaProduct const& product) {
 			return product.m_validJets.size() >= 6 ? (product.m_validJets.at(5)->p4.Pt() >= settings.GetJetOfflineLowerPtCut() ? product.m_validJets.at(5)->p4.mass() : DefaultValues::UndefinedFloat) : DefaultValues::UndefinedFloat;
 		});
+
+        // Lucas' remaining jets custom variable
+        // get the momentum of all jets but the first two
+        
+        LambdaNtupleConsumer<KappaTypes>::AddFloatQuantity("remainingJetPt", [settings](KappaEvent const& event, KappaProduct const& product) {
+                ROOT::Math::LorentzVector<ROOT::Math::PxPyPzE4D<double> > remaining_jets_p4 (0, 0, 0, 0);
+                if (product.m_validJets.size() >= 3) {
+                    typename std::vector<TValidJet*>::const_iterator jet = (product.m_validJets).begin();
+                    std::advance(jet, 2); // ignore first two jets
+                    for (; jet != (product.m_validJets).end(); ++jet) {
+                        if ((*jet)->p4.Pt() >= settings.GetJetOfflineLowerPtCut()) {
+                            remaining_jets_p4 += (*jet)->p4 ;
+                        };
+                    };
+                };
+                return remaining_jets_p4.Pt() ;
+            });
+
+        LambdaNtupleConsumer<KappaTypes>::AddFloatQuantity("remainingJetEta", [settings](KappaEvent const& event, KappaProduct const& product) {
+                ROOT::Math::LorentzVector<ROOT::Math::PxPyPzE4D<double> > remaining_jets_p4 (0, 0, 0, 0);
+                if (product.m_validJets.size() >= 3) {
+                    typename std::vector<TValidJet*>::const_iterator jet = (product.m_validJets).begin();
+                    std::advance(jet, 2); // ignore first two jets
+                    for (; jet != (product.m_validJets).end(); ++jet) {
+                        if ((*jet)->p4.Pt() >= settings.GetJetOfflineLowerPtCut()) {
+                            remaining_jets_p4 += (*jet)->p4 ;
+                        };
+                    };
+                };
+                return remaining_jets_p4.Eta() ;
+            });
+
+        LambdaNtupleConsumer<KappaTypes>::AddFloatQuantity("remainingJetPhi", [settings](KappaEvent const& event, KappaProduct const& product) {
+                ROOT::Math::LorentzVector<ROOT::Math::PxPyPzE4D<double> > remaining_jets_p4 (0, 0, 0, 0);
+                if (product.m_validJets.size() >= 3) {
```

Comment:
maybe you should set a better default value than 0, usually we use -10 here

Reply:
for all the variables so pt,eta,phi

Reply:
Does [this commit](https://github.com/KIT-CMS/Artus/pull/25/commits/934be376eef5144d932d961ae59700b8a1af68ea) makes it correct for you?

Reply:
I just rememberd, for consistency you can use the defaults defined here: https://github.com/KIT-CMS/Artus/blob/reduced_trigger_objects/Utility/src/DefaultValues.cc

Reply:
Ok I updated the code accordingly and tested it.
Note that, if there are more than 2 jets but the remaining ones have pT lower than the `GetJetOfflineLowerPtCut` threshold then you can end up with `Njets_r == 0` and `jpt_r == 0` as well with the current code, as can be seen from [this condition](https://github.com/lucastorterotot/Artus/blob/23dbdec2ac79987c44cf5cd3ad605a983d034a76/KappaAnalysis/interface/Producers/ValidJetsProducer.h#L200-L202).
Would it be better to check this pT threshold only after the sum to only have `jpt_r == -10` for `Njet_r == 0`?